feat(policy): tweak AI policy (#840)
leanprover-community/leanprover-community.github.io: templates/contribute/index.md

Note that using an LLM when writing comments on github or Zulip is not allowed: use your own words.

Using artificial intelligence tools to generate code is becoming more and more common. While this can be practical, their use also poses ethical, ecological, legal and social concerns. We recognise that there are strong differences in opinion on this topic. That said, while individual action alone will not address these concerns, we ask you to consider the effects of your AI use. When reviewing PRs, we are particularly concerned that the pedagogical value of the reviewers work is wasted if there is not a human contributor actively learning.

> Mathlib intentionally has very high standards (on generality, integration with the remaining library and maintainability, including code style). As of mid-2026, AI-written code fails to meet that bar by a large margin. Members of the review team will summarily close without comment any low quality PR produced using LLMs, especially if the author has made little effort to directly engage in the community in a discussion about its merits before opening the PR.

Author: ```suggestion
Mathlib intentionally has very high standards (on generality, integration with the remaining library and maintainability, including code style). As of mid-2026, code written by an AI without the supervision of a Lean subject expert fails to meet that bar by a large margin. Members of the review team will summarily close without comment any low quality PR produced using LLMs, especially if the author has made little effort to directly engage in the community in a discussion about its merits before opening the PR.
```

> Using artificial intelligence tools to generate code is becoming more and more common. While this can be practical, their use also poses ethical, ecological, legal and social concerns. We recognise that there are strong differences in opinion on this topic. That said, while individual action alone will not address these concerns, we ask you to consider the effects of your AI use and if it is truly necessary.

Author: For completeness, this paragraph was also discussed on zulip:
```suggestion
Using artificial intelligence tools to generate code is becoming more and more common. While this can be practical, their use also poses ethical, ecological, legal and social concerns. We recognise that there are strong differences in opinion on this topic. That said, while individual action alone will not address these concerns, we ask you to consider the effects of your AI use. When reviewing PRs, we are particularly concerned that the pedagogical value of the reviewers work is wasted if there is not a human contributor actively learning.
```
other suggestion
```suggestion
Consider if your use of AI is appropriate. There are both positive and negative effects of using AI in your work. When reviewing PRs, we are particularly concerned that the pedagogical value of the reviewers work is wasted if there is not a human contributor actively learning.
```

Author: Zulip seems positive on the first variant, so let me merge that one.

> Mathlib intentionally has very high standards (on generality, integration with the remaining library and maintainability, including code style). As of mid-2026, AI-written code fails to meet that bar by a large margin. Members of the review team reserve the right to summarily close without comment any low quality PR, especially if the author has made little effort to directly engage in the community in a discussion about its merits before opening the PR.

Reviewer: Suggest just "will summarily close" rather than "reserve the right to summarily close".

Author: This would be a policy change (as it would also apply to human-generated PRs). I may be in favour, but think this should be discussed more broadly first. Let me start a thread within the maintainers ([done](https://leanprover.zulipchat.com/#narrow/channel/180721-mathlib-maintainers/topic/Closing.20.2A.2Aany.2A.2A.20low-quality.20PR.3F/with/593475374)).

Author: How about this phrasing instead, suggested by Fillipo? (I like it!)
```suggestion
Mathlib intentionally has very high standards (on generality, integration with the remaining library and maintainability, including code style). As of mid-2026, AI-written code fails to meet that bar by a large margin. Members of the review team will summarily close without comment any low quality PR produced using LLMs, especially if the author has made little effort to directly engage in the community in a discussion about its merits before opening the PR.
```

> Mathlib intentionally has very high standards (on generality, integration with the remaining library and maintainability, including code style). As of mid-2026, AI-written code fails to meet that bar by a large margin. Members of the review team reserve the right to summarily close without comment any low quality PR, especially if the author has made little effort to directly engage in the community in a discussion about its merits before opening the PR.

Reviewer: Suggest replacing

"As of mid-2026, AI-written code fails to meet that bar by a large margin."

with

"As of mid-2026, code written by an AI without the supervision of a Lean subject expert fails to meet that bar."

Author: Given the slop I've seen, I think that is too friendly a phrasing.
(But indeed, the purpose of the sentence is not to say anything like "Kim's AI-generated PRs are crap"; that would be very false.)

Author: I've gone with a compromise, keeping "large" but adding the human expert.

> If we notice that you open several PRs without putting in this learning effort and without adhering to our community ethical standards, we retain the right of banning you both from opening new PR's and from the Zulip chat.

Reviewer: ```suggestion
If we notice that you open several PRs without putting in this learning effort or without adhering to our community ethical standards, we retain the right of banning you both from opening new PR's and from the Zulip chat.
```
?

Reviewer: I thought we had decided that we don't ban, but rather that we use the exponential suspension mechanism? At least for the Zulip, I think we should stick with this. I don't mind so much on the PR front.

Reviewer: And again, I think the "retain the right" language is needlessly legalistic, and it's better to just be declarative.

Author: Thanks! I agree about the "or", and agree that we should mostly keep the suspension system. (In extreme cases, we should still ban directly, but that's our current policy.)

I like being somewhat flexible in our boundaries; as we're discovering, reality turns out more complicated than we thought. (Are we able to suspend users from github for a temporary time, for instance? I don't know!) Perhaps the current phrasing is fine.

Author: How about this variant; does that capture all this point?
```suggestion
If we notice that you open several PRs without putting in this learning effort or without adhering to our community ethical standards, we will suspend (or permanently ban) you both from opening new PR's and from the Zulip chat.
```
